### ISSUE TITLE
why my finetune  T2I model is just 3.7G, is it suitable to generate pics?

### ISSUE THREAD

**Issue by godwenbin** [ByteDance-Seed/Bagel#110]
Hello, I use the training script provided by the repository to train the bagel model. I want to complete the T2I task, that is, only train the T2I part. I have set visual_und=False and successfully trained, but my training checkpoints are only 3.7G, which is much smaller than the original model. Is this in line with expectations? And is this 3.7G model suitable for inference?

![Image](https://github.com/user-attachments/assets/8e32cc3b-ce68-4d16-b2ca-f9e97fd3155d)

![Image](https://github.com/user-attachments/assets/868df73f-467d-4818-b857-74b17bea3981)

**Comment by Andy1621** [comment]
It seems that you are training a ~0.5B model from scratch. The MSE loss is about 2  with randomly initialized.

**Comment by godwenbin** [comment]
@Andy1621 this is my model config:
@dataclass
class ModelArguments:
    model_path: str = field(
        default="/mnt/nj-dev-data-image-text2imge-basemod/safetensors/BAGEL-7B-MoT/",
        metadata={"help": "Path of the pretrained BAGEL model."}
    )
    llm_path: str = field(
        default="/mnt/nj-dev-data-image-text2image-sdb/models/Qwen2.5-0.5B-Instruct/",
        metadata={"help": "Path or HuggingFace repo ID of the pretrained Qwen2-style language model."}
    )
    llm_qk_norm: bool = field(
        default=True,
        metadata={"help": "Enable QK LayerNorm (qk_norm) inside the attention blocks."}
    )
    tie_word_embeddings: bool = field(
        default=False,
        metadata={"help": "Share input and output word embeddings (tied embeddings)."}
    )
    layer_module: str = field(
        default="Qwen2MoTDecoderLayer",
        metadata={"help": "Python class name of the decoder layer to instantiate."}
    )
    vae_path: str = field(
        default="/mnt/nj-dev-data-image-text2imge-basemod/safetensors/BAGEL-7B-MoT/ae.safetensors",
        metadata={"help": "Path to the pretrained VAE checkpoint for latent-space image generation."}
    )
    vit_path: str = field(
        default="/mnt/nj-dev-data-image-text2image-sdb/models/siglip-so400m-14-980-flash-attn2-navit/",
        metadata={"help": "Path or repo ID of the SigLIP Vision Transformer used for image understanding."}
    )
and so what's the key point i can replace to make my  t2i-tranning task work. And can you tell me  how to load this 3.7G model to successfully generate pics?

**Comment by Andy1621** [comment]
Hi! Please provide you training script.

**Comment by godwenbin** [comment]
@Andy1621 Ofcourse! This is my training script:
```
trainning bash:
export WANDB_DISABLED=true

num_nodes=1
node_rank=0
nproc_per_node=8
master_addr=0.0.0.0
master_port=1239
model_path="/mnt/nj-dev-data-image-text2imge-basemod/safetensors/BAGEL-7B-MoT/"
results_dir=results_250528
checkpoint_dir=results_250528/checkpoints

torchrun \
  --nnodes=$num_nodes \
  --node_rank=$node_rank \
  --nproc_per_node=$nproc_per_node \
  --master_addr=$master_addr \
  --master_port=$master_port \
  train/pretrain_unified_navit.py \
  --dataset_config_file ./data/configs/example.yaml \
  --model_path $model_path \
  --layer_module Qwen2MoTDecoderLayer \
  --max_latent_size 64 \
  --finetune_from_hf False \
  --log_every 1 \
  --lr 2e-5 \
  --num_worker 1 \
  --expected_num_tokens 10240 \
  --max_num_tokens 11520 \
  --max_num_tokens_per_sample 10240 \
  --results_dir $results_dir         \
  --checkpoint_dir $checkpoint_dir  \
```
my data/configs/example.yaml
```
t2i_pretrain:
  dataset_names:
  - t2i
  image_transform_args:
    image_stride: 16
    max_image_size: 1024
    min_image_size: 512
  is_mandatory: true
  num_used_data: # The sum should be larger that NUM_GPUS x NUM_WORKERS
  - 10
  weight: 1

data_info.py:
DATASET_INFO = {
    't2i_pretrain': {
        't2i': {
            'data_dir': '/mnt/nj-dev-data-image-text2image-sdb/dataset/1224_dataset_flux_parquet', # path of the parquet files
            'num_files': 1, # number of data units to be sharded across all ranks and workers
            'num_total_samples': 2807, # number of total samples in the dataset
        },
    },
```

**Comment by Andy1621** [comment]
Okay. It seems that you are training a 0.5B LLM, thus the results is normal. If you want to finetune BAGEL, please try the script [here](https://github.com/ByteDance-Seed/Bagel/blob/main/TRAIN.md#training)

```shell
# Fine-tuning
torchrun \
  --nnodes=$num_nodes \
  --node_rank=$node_rank \
  --nproc_per_node=8 \
  --master_addr=$master_addr \
  --master_port=$master_port \
  train/pretrain_unified_navit.py \
  --dataset_config_file ./data/configs/example.yaml \
  --model_path $model_path \
  --layer_module Qwen2MoTDecoderLayer \
  --max_latent_size 64 \
  --resume-from $model_path \
  --finetune_from_hf True \
  --auto_resume True \
  --resume-model-only True \
  --finetune-from-ema True \
  --log_every 1 \
  --lr 2e-5 \
  --num_worker 1 \
  --expected_num_tokens 10240 \
  --max_num_tokens 11520 \
  --max_num_tokens_per_sample 10240
```

**Comment by godwenbin** [comment]
@Andy1621 hello, i don't know what's my script's difference with your, and when i try --resume-from $model_path, i always meet this error:
```
2025-06-04T06:06:23.870431737Z [rank0]: 	size mismatch for language_model.lm_head.weight: copying a param with shape torch.Size([152064, 3584]) from checkpoint, the shape in current model is torch.Size([151936, 896]).
2025-06-04T06:06:23.870434012Z [rank0]: 	size mismatch for time_embedder.mlp.0.weight: copying a param with shape torch.Size([3584, 256]) from checkpoint, the shape in current model is torch.Size([896, 256]).
2025-06-04T06:06:23.870436268Z [rank0]: 	size mismatch for time_embedder.mlp.0.bias: copying a param with shape torch.Size([3584]) from checkpoint, the shape in current model is torch.Size([896]).
2025-06-04T06:06:23.870438489Z [rank0]: 	size mismatch for time_embedder.mlp.2.weight: copying a param with shape torch.Size([3584, 3584]) from checkpoint, the shape in current model is torch.Size([896, 896]).
2025-06-04T06:06:23.870440685Z [rank0]: 	size mismatch for time_embedder.mlp.2.bias: copying a param with shape torch.Size([3584]) from checkpoint, the shape in current model is torch.Size([896]).
2025-06-04T06:06:23.870442881Z [rank0]: 	size mismatch for vae2llm.weight: copying a param with shape torch.Size([3584, 64]) from checkpoint, the shape in current model is torch.Size([896, 64]).
2025-06-04T06:06:23.870445054Z [rank0]: 	size mismatch for vae2llm.bias: copying a param with shape torch.Size([3584]) from checkpoint, the shape in current model is torch.Size([896]).
2025-06-04T06:06:23.870447274Z [rank0]: 	size mismatch for llm2vae.weight: copying a param with shape torch.Size([64, 3584]) from checkpoint, the shape in current model is torch.Size([64, 896]).
```
this is my new script after following yours:
```
export WANDB_DISABLED=true

num_nodes=1
node_rank=0
nproc_per_node=8
master_addr=0.0.0.0
master_port=1239
model_path="/mnt/nj-dev-data-image-text2imge-basemod/safetensors/BAGEL-7B-MoT/"
results_dir=results_250604
checkpoint_dir=results_250604/checkpoints

torchrun \
  --nnodes=$num_nodes \
  --node_rank=$node_rank \
  --nproc_per_node=8 \
  --master_addr=$master_addr \
  --master_port=$master_port \
  train/pretrain_unified_navit.py \
  --dataset_config_file ./data/configs/example.yaml \
  --model_path $model_path \
  --layer_module Qwen2MoTDecoderLayer \
  --max_latent_size 64 \
  --resume-from $model_path \
  --finetune_from_hf False \
  --auto_resume True \
  --resume_model_only True \
  --finetune_from_ema True \
  --log_every 1 \
  --lr 2e-5 \
  --num_worker 1 \
  --expected_num_tokens 10240 \
  --max_num_tokens 11520 \
  --max_num_tokens_per_sample 10240   \
  --results_dir $results_dir         \
  --checkpoint_dir $checkpoint_dir  \
```

**Comment by Andy1621** [comment]
Hi! Since your `finetune_from_hf=False`, it will not load the HF model.

**Comment by godwenbin** [comment]
@Andy1621 yes,because my cloud machine can't catch the HF network, so i need to download the model offline to make my training work. So, what's the HF model's name? is it ${model_path}?i have already download it to use.

**Comment by Andy1621** [comment]
Yes, it is model_path

**Comment by godwenbin** [comment]
@Andy1621 ok, i know. But i meet  the size-mismatch error after using your sample script...

**Comment by godwenbin** [comment]
@Andy1621 I have run successfully this time and help check why my Train Loss ce is always 0.
```
2025-06-04T12:19:07.175318195Z Yielding data with length 9350
2025-06-04T12:19:08.904554817Z [[34m2025-06-04 12:19:08[0m] (step=0000143) Train Loss mse: 0.4200, Train Loss ce: 0.0000, Train Steps/Sec: 0.42, 
2025-06-04T12:19:09.261684099Z Yielding data with length 9822
2025-06-04T12:19:09.314644337Z Yielding data with length 9822
2025-06-04T12:19:09.317026662Z Yielding data with length 9755
2025-06-04T12:19:09.317914028Z Yielding data with length 9755
2025-06-04T12:19:09.331152640Z Yielding data with length 9822
2025-06-04T12:19:09.349503765Z Yielding data with length 9822
2025-06-04T12:19:09.360309530Z Yielding data with length 9822
2025-06-04T12:19:09.373519376Z Yielding data with length 9822
2025-06-04T12:19:11.122175885Z [[34m2025-06-04 12:19:11[0m] (step=0000144) Train Loss mse: 0.4225, Train Loss ce: 0.0000, Train Steps/Sec: 0.45, 
2025-06-04T12:19:13.338962790Z [[34m2025-06-04 12:19:13[0m] (step=0000145) Train Loss mse: 0.4938, Train Loss ce: 0.0000, Train Steps/Sec: 0.45, 
2025-06-04T12:19:15.626273124Z [[34m2025-06-04 12:19:15[0m] (step=0000146) Train Loss mse: 0.4424, Train Loss ce: 0.0000, Train Steps/Sec: 0.44, 
2025-06-04T12:19:18.125225441Z [[34m2025-06-04 12:19:18[0m] (step=0000147) Train Loss mse: 0.3903, Train Loss ce: 0.0000, Train Steps/Sec: 0.40, 
2025-06-04T12:19:20.646607160Z [[34m2025-06-04 12:19:20[0m] (step=0000148) Train Loss mse: 0.4103, Train Loss ce: 0.0000, Train Steps/Sec: 0.40, 
```

**Comment by Andy1621** [comment]
Since you set `visual_und=False`, the ce_loss will be 0.
